components/org.wso2.carbon.identity.oauth.endpoint/src/test/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtilTest.java

```diff
@@ -342,10 +355,16 @@ public void testGetUserConsentURL(Object oAuth2ParamObject, boolean isOIDC, bool
                     "is not found in url");
             Assert.assertTrue(ArrayUtils.contains(scopeArray, "internal_login"), "internal_login " +
                     "scope parameter value is not found in url");
+
             if (queryString != null && cacheEntryExists) {
                 Assert.assertTrue(consentUrl.contains(queryString), "spQueryParams value is not found in url");
             }
 
+            if (parameters.getScopes().contains("openid")) {
+                Assert.assertTrue(URLDecoder.decode(consentUrl, "UTF-8").contains(REQUESTED_OIDC_SCOPES),
```

Comment:
Shall we get the `requested_oidc_scopes` param and assert whether the exact values we send is present in the format that it is expected.